### DISCUSSION TITLE
`xsel` hangs when called from a i3 binding?

### DISCUSSION THREAD

**Discussion by ldorigo** [i3/i3#4929] [Q&A]
Hi, I have a very simple (testing) script:

```bash
#! /usr/bin/bash

echo "testing" >> /tmp/test.log;
xsel  >> /tmp/test.log 2>&1;
echo "done" >> /tmp/test.log;
```
When I run it from the terminal, it works as expected - `/tmp/test.log` contains `testing <selection> done`. When I run this script via a keybinding, however, the script hangs on `xsel` - `/tmp/test.log` only contains `testing <selection>`, the last `echo` is never executed. Why is that?

Relevant config line:

```
        bindsym e exec /path/to/test.sh ; mode "default"
```

**Comment by Airblader** [comment]
Try using `bindsym --release`.

**Reply by ldorigo** [reply]
As in `bindsym --release e exec ...`? Same result.
